staging/src/k8s.io/dynamic-resource-allocation/structured/pools.go

```diff
@@ -101,17 +104,32 @@ func addSlice(pools map[PoolID]*Pool, slice *resourceapi.ResourceSlice) {
 
 	if slice.Spec.Pool.Generation > pool.Slices[0].Spec.Pool.Generation {
 		// Newer, replaces all old slices.
-		pool.Slices = []*resourceapi.ResourceSlice{slice}
+		pool.Slices = nil
 	}
 
 	// Add to pool.
 	pool.Slices = append(pool.Slices, slice)
 }
 
+func poolIsInvalid(pool *Pool) (bool, string) {
```

Comment:
Is the only scenario where the pool is invalid when the device name is duplicated? Are there other scenarios to consider? Length limit?

Reply:
The length limit is enforced by the apiserver, so that can't occur. What the apiserver cannot validate are constraints across multiple objects, therefore the combined pool might be invalid.

I can't think of any other aspect that is worth checking here. The "multiple devices" one was something that came out of the observation that a consumer might see such duplicated slices if the producer malfunctions.